src/main/scala/regmapper/RegField.scala

```diff
@@ -7,6 +7,37 @@ import chisel3.util.{ReadyValidIO}
 
 import freechips.rocketchip.util.{SimpleRegIO}
 
+
+// This information is not used internally by the regmap(...) function.
+// However, the author of a RegField may be the best person to provide this
+// information which is likely to be needed by downstream SW and Documentation
+// tools.
+
+object RegFieldAccessType extends scala.Enumeration {
+  type RegFieldAccessType = Value
+  val R, W, RW, RSPECIAL, WSPECIAL, RWSPECIAL, OTHER = Value
+}
+import RegFieldAccessType._
+
+
+object RegFieldResetType extends scala.Enumeration {
+  type RegFieldResetType = Value
+  val S, A, N = Value
```

Comment:
Would it be appropriate to elaborate in a comment here what S, A and N stand for?

Reply:
done